onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -3257,10 +3257,65 @@ def aten_matrix_power(self: TensorType, n: int) -> TensorType:
     raise NotImplementedError()
 
 
-def aten_max(self: TensorType) -> TensorType:
+@torch_op("aten::max", trace_only=True)
+def aten_max(self: TReal, dim_or_other: TReal = None, keepdim: BOOL = None) -> TReal:
     """max(Tensor self) -> Tensor"""
 
-    raise NotImplementedError()
+    self_rank = op.Size(op.Shape(self))
+    if self_rank == 0:
+        self = op.Reshape(self, op.Constant(value_int=[-1]))
+
+    output = 1
+
+    if dim_or_other is None and keepdim is None:
+        result = _aten_max_with_no_dim(self)
+    elif keepdim is None:
+        if isinstance(dim_or_other, int):
```

Comment:
if dim_or_other could be int, it should be updated in annotation

Reply:
done.